File: implementation/scipamato/core/web/src/test/java/ch/difty/scipamato/core/web/model/NewsletterTopicModelTest.java
package ch.difty.scipamato.core.web.model;

import static org.assertj.core.api.Assertions.assertThat;
import static org.mockito.Mockito.*;

import java.util.ArrayList;
import java.util.List;

import org.junit.jupiter.api.Test;

import ch.difty.scipamato.core.entity.newsletter.NewsletterTopic;

class NewsletterTopicModelTest extends ModelTest {

    @Test
    void loading_delegatesToNewsletterTopicService() {
        String languageCode = "de";

        final List<NewsletterTopic> topics = new ArrayList<>();
        topics.add(new NewsletterTopic(1, "t1"));
        topics.add(new NewsletterTopic(2, "t2"));

        when(newsletterTopicServiceMock.findAll(languageCode)).thenReturn(topics);

        final NewsletterTopicModel model = new NewsletterTopicModel("de");

        assertThat(model.load())
            .extracting(NewsletterTopic.NewsletterTopicFields.TITLE.getFieldName())
            .containsExactly("t1", "t2");

        verify(newsletterTopicServiceMock).findAll(languageCode);

        verifyNoMoreInteractions(newsletterTopicServiceMock);
    }
}

File: core/core-web/src/test/kotlin/ch/difty/scipamato/core/web/model/NewsletterTopicModelTest.kt
package ch.difty.scipamato.core.web.model

import ch.difty.scipamato.core.entity.newsletter.NewsletterTopic
import io.mockk.confirmVerified
import io.mockk.every
import io.mockk.verify
import org.amshove.kluent.shouldContainSame
import org.junit.jupiter.api.Test

internal class NewsletterTopicModelTest : ModelTest() {

    @Test
    fun loading_delegatesToNewsletterTopicService() {
        val languageCode = "de"
        val topics = listOf(NewsletterTopic(1, "t1"), NewsletterTopic(2, "t2"))
        every { newsletterTopicServiceMock.findAll(languageCode) } returns topics
        val model = NewsletterTopicModel("de")

        model.load().map { it.title } shouldContainSame listOf("t1", "t2")

        verify { newsletterTopicServiceMock.findAll(languageCode) }
        confirmVerified(newsletterTopicServiceMock)
    }
}
